daemons/fenced/fenced_commands.c

```diff
@@ -428,6 +428,14 @@ stonith_device_execute(stonith_device_t * device)
         cmd->activating_on = NULL;
         cmd->done_cb(0, exec_rc, NULL, cmd);
     }
+
+done:
+    /* Device might get triggered to work by multiple fencing commands
+     * simultaneously. Trigger the device again to make sure any
+     * remaining concurrent commands get executed. */
+    if (device->pending_ops && g_list_length(device->pending_ops) > 0) {
```

Comment:
Not worth changing, especially since this list will never be very long, but "if (device->pending_ops)" is sufficient here. An empty GList* is guaranteed to be NULL, and a non-empty list non-NULL (unlike GHashTable*, where NULL is invalid and an empty table is non-NULL).

Reply:
Makes sense too.